backend-shared/db/application_test.go

```diff
@@ -145,4 +158,43 @@ var _ = Describe("Application Test", func() {
 		})
 	})
 
+	Context("Test ListApplicationsForManagedEnvironment function", func() {
```

Comment:
See my earlier comment regarding the coverage of the if statement.  Similar solution should apply here.



```
if err != nil {
  return 0, fmt.Errorf("unable to retrieve applications with managed environment id: %v", err)
}
```